src/Agent.Listener/SelfUpdater.cs

```diff
@@ -179,7 +188,7 @@ private async Task DownloadLatestAgent(CancellationToken token)
             try
             {
                 // Download the agent, using multiple attempts in order to be resilient against any networking/CDN issues
-                for (int attempt = 1; attempt <= Constants.AgentDownloadRetryMaxAttempts; attempt++)
+                for (int attempt = 1; attempt <= Constants.AgentDownloadRetryMaxAttempts && !downloadSucceeded; attempt++)
                 {
                     // Generate an available package name, and do our best effort to clean up stale local zip files
                     while (true)
```

Comment:
Where could be infinite loop for the case when exception will happen inside of inner try.. catch

Reply:
Here we generate an available package name. The only reason why we have a `try...catch` block is to avoid the situation when there already is a package with the generated name and it cannot be deleted — then we just generate another available package name by incrementing its suffix. This is equivalent to the following code:
```ts
// Generate an available package name, and do our best effort to clean up stale local zip files
bool packageNameGenerated = false;
while (!packageNameGenerated)
{
    if (_targetPackage.Platform.StartsWith("win"))
    {
        archiveFile = Path.Combine(latestAgentDirectory, $"agent{agentSuffix}.zip");
    }
    else
    {
        archiveFile = Path.Combine(latestAgentDirectory, $"agent{agentSuffix}.tar.gz");
    }

    // delete .zip file
    if (!string.IsNullOrEmpty(archiveFile) && File.Exists(archiveFile))
    {
        Trace.Verbose("Deleting latest agent package zip '{0}'", archiveFile);
        try
        {
            IOUtil.DeleteFile(archiveFile);
            packageNameGenerated = true;
        }
        catch (Exception ex)
        {
            // couldn't delete the file for whatever reason, so generate another name
            Trace.Warning("Failed to delete agent package zip '{0}'. Exception: {1}", archiveFile, ex);
            agentSuffix++;
        }
    }
    else
    {
        packageNameGenerated = true;
    }
}
```
Is this a more readable solution?